### ISSUE TITLE
Not sure `--host` is working with https://test.pypi.org?

### ISSUE THREAD

**Issue by lorengordon** [arcivanov/pypi-cleanup#1]
Hi, this looks super handy! I just ran into the pypi size limit with a pkg we've been publishing to https://test.pypi.org for years... It's a lot of old dev pkgs to cleanup, so went looking for help and found this project!

However, it seems to fail on authentication to test.pypi.org... I know the credential is fine, since I can login no problem using the browser. Any idea? At first glance, the code seems fine, but is failing this logic:

https://github.com/arcivanov/pypi-cleanup/blob/e82676c8c703d150deba41dcad109c643e7ae2ed/src/main/python/pypi_cleanup/__init__.py#L134-L136

```
pypi-cleanup -p watchmaker -u xxxxxxxxxx --host https://test.pypi.org/ -n -r '0\.0\..*' -v
WARNING:root:RUNNING IN DRY-RUN MODE
INFO:root:Will use the following patterns [re.compile('0\\.0\\..*')] on package watchmaker
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): pypi.python.org
DEBUG:urllib3.connectionpool:https://pypi.python.org:443 "GET /pypi/watchmaker/json HTTP/1.1" 301 122
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): pypi.org
DEBUG:urllib3.connectionpool:https://pypi.org:443 "GET /pypi/watchmaker/json HTTP/1.1" 200 34240
Password:
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): test.pypi.org
DEBUG:urllib3.connectionpool:https://test.pypi.org:443 "GET /account/login/ HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://test.pypi.org:443 "POST /account/login/ HTTP/1.1" 200 None
ERROR:root:Login for user xxxxxxxxxx failed
```

**Comment by arcivanov** [comment]
Glad it helps. Thanks, let me take a look! I've actually never tried it with non-default host.

**Comment by arcivanov** [comment]
The cause of this is: https://github.com/arcivanov/pypi-cleanup/blob/e82676c8c703d150deba41dcad109c643e7ae2ed/src/main/python/pypi_cleanup/__init__.py#L81

**Comment by arcivanov** [comment]
Since you're in a dry-run mode, have you tried with the default host?

**Comment by lorengordon** [comment]
Using the default host does indeed seem to work!

**Comment by arcivanov** [comment]
I suspect there is a site structure difference in test.

**Comment by arcivanov** [comment]
Also, your account in test is generally different from your primary one.

**Comment by lorengordon** [comment]
It is a different account, but I understand that and am using the correct account for each site. Other than the url, the site structure seems to be the same. But maybe the response is different in some way. I'll install from source and add some extra logging to see if I can figure it out.

**Comment by lorengordon** [comment]
Ok, different problem. Login works fine if I run from source. So maybe what is published to pypi is different from what is in the repo. Of course, it's not finding the packages, since it is looking at the default host for those.

```
$ python src/main/python/pypi_cleanup/__init__.py -p watchmaker -u plus3it -t https://test.pypi.org -n -v
WARNING:root:RUNNING IN DRY-RUN MODE
INFO:root:Will use the following patterns [re.compile('.*\\.dev\\d+$')] on package watchmaker
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): pypi.python.org
DEBUG:urllib3.connectionpool:https://pypi.python.org:443 "GET /pypi/watchmaker/json HTTP/1.1" 301 122
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): pypi.org
DEBUG:urllib3.connectionpool:https://pypi.org:443 "GET /pypi/watchmaker/json HTTP/1.1" 200 34240
Password:
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): test.pypi.org
DEBUG:urllib3.connectionpool:https://test.pypi.org:443 "GET /account/login/ HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://test.pypi.org:443 "POST /account/login/ HTTP/1.1" 303 198
DEBUG:urllib3.connectionpool:https://test.pypi.org:443 "GET /manage/projects/ HTTP/1.1" 200 4863
INFO:root:Deleting watchmaker version 0.2.3.dev0
```

**Comment by arcivanov** [comment]
That's unlikely but I'll check.

**Comment by arcivanov** [comment]
Is it possible your installed environment differs from your development environment?

**Comment by arcivanov** [comment]
> I just ran into the pypi size limit 

Speaking of which, could you please share what the limits are? Are there limits on PyPI prod and what are they?

**Comment by lorengordon** [comment]
Here's the error we are currently getting when posting to test.pypi.org, indicates the project has a limit of 10 GB... I think it is a per-project setting, and can be increased by request.

> HTTPError: 400 Client Error: Project size too large. Limit for project '[secure]' total size is 10 GB. See https://test.pypi.org/help/#project-size-limit for url: https://test.pypi.org/legacy/

**Comment by arcivanov** [comment]
Thanks!

**Comment by lorengordon** [comment]
Changed the source to get packages from `self.url` and hit a new error where the post to delete the pkg 404s. Tested on the default host with the one dev pkg we accidentally released there years ago and reproduced the error on the default host...

```
$ python src/main/python/pypi_cleanup/__init__.py -p watchmaker -u plus3it -v
INFO:root:Will use the following patterns [re.compile('.*\\.dev\\d+$')] on package watchmaker
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): pypi.org
DEBUG:urllib3.connectionpool:https://pypi.org:443 "GET /pypi/watchmaker/json HTTP/1.1" 200 34240
Password:
DEBUG:urllib3.connectionpool:https://pypi.org:443 "GET /account/login/ HTTP/1.1" 200 None
DEBUG:urllib3.connectionpool:https://pypi.org:443 "POST /account/login/ HTTP/1.1" 303 198
DEBUG:urllib3.connectionpool:https://pypi.org:443 "GET /manage/projects/ HTTP/1.1" 200 4928
INFO:root:Deleting watchmaker version 0.2.3.dev0
DEBUG:urllib3.connectionpool:https://pypi.org:443 "GET /manage/project/watchmaker/releases/ HTTP/1.1" 200 20399
DEBUG:urllib3.connectionpool:https://pypi.org:443 "POST /manage/project/watchmaker/release/0.2.3.dev0/ HTTP/1.1" 404 None
Traceback (most recent call last):
  File "src/main/python/pypi_cleanup/__init__.py", line 213, in <module>
    sys.exit(main())
  File "src/main/python/pypi_cleanup/__init__.py", line 209, in main
    return PypiCleanup(**vars(args)).run()
  File "src/main/python/pypi_cleanup/__init__.py", line 177, in run
    r.raise_for_status()
  File "/home/loren/.local/lib/python3.6/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://pypi.org/manage/project/watchmaker/release/0.2.3.dev0/
```

**Comment by arcivanov** [comment]
Hmm, you don't seem to have TOTP on your account...

**Comment by lorengordon** [comment]
That's true, we automate the releases, and I haven't yet switched to the newish token method for that.

**Comment by arcivanov** [comment]
Let me try it on my stuff and see if I get the same issue. PyPI might've changed the way they authenticate so there may be a need to change the scraper.

**Comment by arcivanov** [comment]
I confirm it isn't working for me either. PyPI must've changed something on their website.

**Comment by arcivanov** [comment]
Yep, they did: "confirm_version" became "confirm_delete_version".

**Comment by lorengordon** [comment]
That did it! Made the change locally and now rockin and rollin

**Comment by arcivanov** [comment]
Version 0.0.2 released. Fixed in #2.

**Comment by lorengordon** [comment]
Thanks @arcivanov, I can confirm 0.0.2 is working!
